src/walletNode.ts

```diff
@@ -275,18 +276,43 @@ export class WalletNode {
       response.content.genesisBlockIdentifier.hash,
       'hex',
     )
-    const walletGenesisHeader = BlockHeaderSerde.deserialize(
-      networkDefinition.genesis.header,
-      this.strategy,
-    )
 
-    if (walletGenesisHeader.hash.equals(nodeGenesisHash)) {
+    let fishHashContext: FishHashContext | undefined = undefined
+    if (!this.network.consensus.isNeverActive('enableFishHash')) {
+      fishHashContext = new FishHashContext(false)
+    }
+
+    const blockHasher = new BlockHasher({
+      consensus: this.network.consensus,
+      context: fishHashContext,
+    })
+
+    const genesisHeader = networkDefinition.genesis.header
```

Comment:
This isn't a header, it's a header definition. If it was a header I expect it to be passable directly into hashHeader